src/Plugins/Internal/Debouncer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace McMaster.NETCore.Plugins.Internal
+{
+    internal class Debouncer : IDisposable
```

Comment:
Can we add some unit tests for this to verify its behavior (and prevent regressions in the future)?

Reply:
I've pushed some unit tests

Reply:
Yes, thank you!